clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy

```diff
@@ -198,15 +201,27 @@ class DockerRegistryClient {
    * This method will hit the /v2/ endpoint of the configured docker registry. If it this endpoint is up,
    * it will return silently. Otherwise, an exception is thrown detailing why the endpoint isn't available.
    */
+
   public void checkV2Availability() {
+    try {
+      doRequest(tokenService.basicAuthHeader)
+    } catch (RetrofitError error) {
+        LOG.warn "checkV2Availability", error
+        Response response = doRequest(null)
+        if(response != null){
+          throw error
+        }
+    }
+    // Placate the linter (otherwise it expects to return the result of `request()`)
+    null
+  }
+
+  private Response doRequest(String basicAuthHeader) {
```

Comment:
Why is the v2 version check the only request using this function? I'd rename the function or apply it to other requests